esterni/manuale-sviluppatore/components/tecnologie-interessate.tex

```diff
@@ -0,0 +1,59 @@
+\documentclass[../manuale-sviluppatore.tex]{subfiles}
+
+\begin{document}
+\subsection{Back End}%
+\label{sub:back-end}
+In questa sezione sono descritte le tecnologie scelte dopo una fase di analisi per lo sviluppo del back-end.
+
+\subsubsection{InfluxDB}%
+\label{sub:influxDB}
+Abbiamo deciso di utilizzare InfluxDB, un Time Series Database open source, viste le sue alte performance e le sue elevate capacità di scrittura e di sostenere carichi di interrogazioni 
+importanti. Usiamo questo database per salvare dati momentanei, non persistenti.
+% TODO estendere
+
+\subsubsection{MySql}%
+\label{sub:mysql}
+Abbiamo deciso di utilizzare MySql per il salvataggio di dati persistenti attraverso il modello relazionale, il quale garantisce affidabilità.
+% TODO estendere
+
+\subsubsection{Spring}%
+\label{sub:spring}
+Per sviluppare il back-end abbiamo deciso di utilizzare Spring, un framework concepito per la realizzazione di applicazioni enterprise, che soddisfa i requisiti
+di performance, sicurezza e affidabilità richiesti. \\
+Spring contiene molte librerie, che ci hanno consentito di risolvere in modo elegante più di qualche requisito.
+Unico lato negativo di questo framework è la sua documentazione, che può risultare molto dispersiva.
+% TODO estendere
+
+\subsubsection{Open API}%
+\label{sub:open_api}
+Grazie alla specifica Open API abbiamo definito le interfacce per la comunicazione, andando quindi a semplificare la sincronizzazione tra documentazione e codice sorgente.
+Open API è in grado di generare codice, documentazione e test case dato un file di interfaccia.
+% TODO estendere
+
+
+\newpage
+\subsection{Web Application}%
+\label{sub:back-end}
+In questa sezione sono descritte le tecnologie scelte dopo una fase di analisi per lo sviluppo dell'applicazione web.
+
+\subsubsection{Angular}%
+\label{sub:angular}
+Abbiamo scelto Angular come framework per lo sviluppo web, visto che consente di creare applicazioni moderne e molto avanzate in modo facile e veloce.
+Alla base dell'architettura Model View ViewModel di Angular c'è TypeScript, necessario a chiunque voglia utilizzare Angular.\\
+I principali tipi di classi in Angular sono i components, cioè la vista, e i servizi, che interagiscono con un modello.
```

Comment:
La forma non va bene, rivedere questa parte utilizzando un linguaggio più appropriato

Reply:
Aggiungo: il commento su typescript è sbagliato. Non serve typescript, basta javascript (anche se serve una versione abbastanza nuova).

Reply:
per lo sviluppo in Angular serve imprescindibilmente typescript, ma non sei "obbligato" a usarlo perché le feature sono un superset di quelle di JS. però il transpiler lo devi usare comunque

Reply:
qualche pointer più specifico su cosa non va bene secondo me:

- TypeScript è un linguaggio di programmazione, non è alla base dell'architettura (perché le architetture sono language-agnostic)
- se vai a leggere la pagina https://angular.io/guide/architecture, dice che i componenti _definiscono_ le view, non _sono_ le view. la differenza è abbastanza grossa.